packages/zoe/src/zoeService/startInstance.js

```diff
@@ -248,13 +265,39 @@ export const makeStartInstance = (
           'The contract did not correctly return a creatorInvitation',
         );
       }
+      if (zcfBundleCapResult !== undefined) {
+        assert(
+          zcfBundleCapResult.status === 'fulfilled' && zcfBundleCapResult.value,
+          'the budnle cap was broken',
+        );
+      }
       const adminFacet = Far('adminFacet', {
         getVatShutdownPromise: () => E(adminNode).done(),
+        restartContract: _newPrivateArgs => {
+          const vatParameters = { contractBundleCap };
+          return E(adminNode).upgrade(zcfBundleCapResult.value, {
```

Comment:
Since `restartContract` doesn't necessarily upgrade, should this be:
```suggestion
          return E(adminNode).restart(zcfBundleCapResult.value, {
```
It would make sense in 290 that `upgradeContract` calls `E(adminNode).restart` after it does upgradey things

Reply:
that's a kernel method (`vat-vat-admin.js`).   @warner   Is this at all plausible?

Reply:
Hm, that's a good question. Our primary use case for this kernel feature is to upgrade a vat, and our documentation describes it in those terms, but yes, the code provided to the new incarnation might be the same (or even older than) the code that was used in the previous incarnation. `upgrade` implies the new code is newer than the previous code, so it's not ideal. But `restart` implies (to me) that the new code is the same as the previous code, which feels slightly worse.

I suppose if I compare english phrases, "Restart, but with a newer version of the code" vs "Upgrade, but with the same version of the code", then using `restart` *does* feel less wrong.

As kernel services go, naming it `upgrade` makes it clear that this can be used for upgrades, whereas naming it `restart` needs extra docs to make it clear that changing the version is a possibility.

If we were to change it, then we'd need to change the API to match: `E(adminNode).restart()` should restart the vat with its old bundlecap, instead of obligating the caller to remember it and re-supply it anew. The `bundlecap` argument would need to move to an option, so "restart" is the primary mode, and "upgrade" is a secondary/optional mode. That's not horrible..

All of our docs are written around the idea of upgrade, rather than restart. I'd probably want to rewrite a bunch, `packages/SwingSet/docs/vat-upgrade.md` would become `vat-upgrade-restart.md`, etc.

I'm guessing it would take a day or so to change the implementation, tests, and docs.

Overall I'm -0 on changing from `upgrade()` to `restart()`, but I could be talked into it by a compelling argument about `restart()` being more meaningful/coherent/accurate.

Reply:
@warner I love your explanation of the pros and cons of these terminology choices. We see this structure of tradeoffs often, but I've never seen words for it before. Clarifying!

I'd add one point: With "upgrade" as the central word and concept, we already find ourselves using a term for "restart with the same code". To whit "null upgrade". By making it an "adjective upgrade" we make it clear that we still consider it a member of the upgrade category, but a member where something normally present is missing (has been nulled).

So I'm +1 on sticking with "upgrade" and "null upgrade"